lib/graphql/type_kinds.rb

```diff
@@ -43,10 +43,36 @@ def composite?; @composite; end
     ]
 
     class TypeKind
-      TYPE_KINDS.map(&:name).each do |kind_name|
-        define_method("#{kind_name.downcase}?") do
-          self.name == kind_name
-        end
+      def scalar?
+        self.name == TypeKinds::SCALAR.name
+      end
```

Comment:
Honestly, I don't know why I did this way the first time (it was 3 years ago, after all :P), what do you think about comparing the _objects_ instead of their names, for example: 

```ruby 
def scalar? 
  self == TypeKinds::SCALAR 
end 
```